hybrid-cloud-patterns/connected-vehicle-architecture
# Cleanup super linter and update common/

- Add new ignores for python module
- Add python module implementing the push-secret functionality
- Switch push_secrets.yaml to the python module implementation
- Add more detail to the values-secret.yaml example
- Fix up tests after the secrets example change
- Tests cleanups
- Some more linting fixes
- Some more linting fixes pt2
- Pass black --skip-string-normalization over the python plugin
- Pass black --skip-string-normalization over ansible unit test
- Switched to full black formatting support
- Add some noqa: comments to ignore flake8
- Add .gitleaks.toml
- Also ignore python modules in gitleaks as they contain secrets examples in the comments
- Move .gitleaks.toml under .github/linters
- Add a CHANGES.md entry for the push_secrets rewrite
- Drop -f example/values-secrets.yaml from make test
- Drop GIT_SSH_COMMAND from the wrapper
- Return {} when parsing an empty yaml file
- Make sure to skip loops that have None in their range
- Make sure to expand user path when checking for file existence
- Add a number of testcases that cover corner cases in the values files
- Fix up flake8 and black warnings
- Fix ansible-lint
- Cleanup tests and bail out properly when files/secrets: are lists
- Reformat isinstance list fail message
- Add a pattern_dir variable to the playbook
- Add a check a check_missing_secrets parameter to vault_load_secrets
- Add a number of unit tests to verify the new check_missing_secrets functionality
- Add all the code to actually verify secrets against values-secret-template.yaml but keep it disabled
- Fix a couple of linting issues
- Tweak the SSH paths so ssh-agent works again
- Pass -e KUBECONFIG *only* if it is set
- Get the correct pattern name when the path has a folder
- Implement global.clusterVersion
- Tweak the /values-ocpversion calculation in the acm policy
- Be more precise with the Changes.md
- Use a simpler expression in ACM templates
- Add a note about argo hub and spoke and clusterversion having to be the same in that case
- Add possibility to retry commands in run_command()
- Symlink .github/linters/.gitleaks.toml to .gitleaks.toml
- Skip git ssh checks when running inside a container
- Lower case cluster names in values-example.yaml
- Switch to a more recent setup-helm action
- Update ESO to 0.6.0
- Use UBI based images for ESO
- Use no_log: true in any sensitive place
- Disable VALIDATE_DOCKERFILE_HADOLINT VALIDATE_TEKTON by default in common/Makefile
- Drop super-linter target, reuse the one in common
